src/orion/core/worker/consumer.py

```diff
@@ -65,10 +72,12 @@ def consume(self, trial):
         :type trial: `orion.core.worker.trial.Trial`
 
         """
-        log.debug("### Create new temporary directory at '%s':", self.tmp_dir)
-        with tempfile.TemporaryDirectory(prefix=self.experiment.name + '_',
-                                         dir=self.tmp_dir) as workdirname:
-            log.debug("## New temp consumer context: %s", workdirname)
+        log.debug("### Create new directory at '%s':", self.dir)
+        temp_dir = self.experiment.working_dir is None
+        prefix = self.experiment.name + "_"
```

Comment:
It should be ```'{experiment.name}_{trial.id}'``` otherwise there will be clash of dirname for different trials when `working_dir` is set. Note that the trailing `_` is only required when `working_dir` is not set since it is appended with a random string.